pkg/collector/horizontalpodautoscaler.go

```diff
@@ -35,18 +35,18 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 
 	return autoscalingv1.HorizontalPodAutoscaler{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        naming.Collector(otelcol),
+			Name:        otelcol.Name,
```

Comment:
What is the rationale for the name change? Could this produce an invalid name?

Reply:
We are now scaling on the OpenTelemetryCollector rather than the deployment.  If the otelcol.Name is "simple" we append "-collector" to the deployment name.  

If you think it's a good idea I could add a function to naming which just does

```
	return DNSName(Truncate("%s", 63, otelcol.Name))
```